### PR TITLE
nezuko: round11 — volume_loss_weight=1.5 (gentler than #142, attack vol_p gap)

### PR DESCRIPTION
**Pull request by morganmcg1**
## Hypothesis

Increase volume loss weight from `volume_loss_weight=1.0` (SOTA) to `volume_loss_weight=1.5` as a clean single-variable delta from PR #115 SOTA. Target the largest physical gap vs AB-UPT: `volume_pressure_rel_l2_pct` is at 12.740 in our SOTA vs 6.08 in AB-UPT (×2.1 — the dominant gap among reported metrics).

The key justification is that thorfinn's PR #142 (`vol_w=2.0`) overshot — it badly hurt surface metrics (+18% surface_pressure regression) without moving volume_pressure. The post-mortem on #142 was that 2.0 is too aggressive and creates a ratio imbalance with the surface loss. This experiment tests the gentler 1.5 weighting: enough to give the volume head more gradient signal, not enough to starve the surface head.

This is also a different mechanism than PR #149 (per-axis tau weights) — which failed because Lion's sign-based updates neutralize per-channel weighting. Volume loss weight is a global multiplier on the volume task vs surface task, which Lion *will* respect at the optimizer level (changes the relative magnitude of surface vs volume sign-updates per step, which is determined by the gradient distribution across parameter blocks even with sign updates).

## Instructions

From the SOTA stack (PR #115: Lion lr=1e-4 + EMA=0.999), **change exactly one flag**: `--volume-loss-weight 1.5` (default is 1.0).

Run exactly:

```bash
cd target/
torchrun --standalone --nproc_per_node=8 train.py \
  --agent nezuko \
  --optimizer lion --lr 1e-4 --weight-decay 5e-4 \
  --lion-beta1 0.9 --lion-beta2 0.99 \
  --no-compile-model --batch-size 4 --validation-every 1 \
  --epochs 9 \
  --train-surface-points 65536 --eval-surface-points 65536 \
  --train-volume-points 65536 --eval-volume-points 65536 \
  --model-layers 4 --model-hidden-dim 512 --model-heads 8 --model-slices 128 \
  --use-ema --ema-decay 0.999 \
  --lr-cosine-t-max 50 \
  --volume-loss-weight 1.5 \
  --wandb-group tay-round11-vol-loss-weight-1p5 \
  --wandb-name nezuko/round11-vol-loss-weight-1p5
```

Note `--lr-cosine-t-max 50` is included to match SOTA's effective LR schedule (PR #115's `t_max=0` falls back to `T_max=epochs=50` in code; specifying 50 explicitly avoids the LR-collapse confound seen in some earlier runs at `t_max=0, epochs=9`).

**Things to watch:**

- **Primary signal**: `val_primary/volume_pressure_rel_l2_pct`. SOTA reaches val vol_p around ~13.5 by ep8. If we see < 12.5 by ep6-7, that's an early winning signal. If vol_p improves while abupt is flat or improved, this is a winner.
- **Risk metric**: `val_primary/surface_pressure_rel_l2_pct`. SOTA reaches ~5.5–6.0 val surface by ep8. If surface val is > 7.0 by ep6, the weighting has tipped over (similar to #142 failure mode). Flag it but don't kill the run — partial-win is informative.
- ep0 val abupt should be ≤ 55 (vs SOTA 53.75). ep1 ≤ 25 (vs SOTA 24.15). Larger deltas would suggest the loss reweighting destabilized early training.
- Reference val trajectory (abupt) PR #115 SOTA: 53.75 / 24.15 / 16.51 / 13.47 / 11.83 / 10.88 / 10.16 / 9.73 / 9.484
- Per-epoch time should match SOTA (~32 min/ep) — this is a loss-only change, not a compute change.

**Do not add any other flags.** Single-variable test.

## Baseline (from BASELINE.md — PR #115, merged)

Current best test metrics (lower is better):

| Metric | PR #115 SOTA | AB-UPT ref | Gap |
|---|---:|---:|---:|
| **abupt_mean** | **10.580** | — | — |
| surface_pressure | 5.690 | 3.82 | ×1.5 |
| wall_shear | 10.419 | 7.29 | ×1.4 |
| **volume_pressure** | **12.740** | 6.08 | **×2.1** ← target |
| tau_x | 8.908 | 5.35 | ×1.7 |
| tau_y | 12.491 | 3.65 | ×3.4 |
| tau_z | 13.071 | 3.63 | ×3.6 |

Best val: 9.484 (ep8). Val→test ratio ~1.115.

W&B run: `d03oghpp` (thorfinn compound lr=1e-4 + EMA=0.999, 287 min, 9 epochs).

## Prior data point — why 1.5 not 2.0

PR #142 (thorfinn, `volume_loss_weight=2.0`):
- abupt_mean: 11.721 (+10.78% regression)
- surface_pressure: 6.71 (+18% worse)
- volume_pressure: 12.69 (essentially flat — no improvement on the target metric)
- Conclusion: 2.0 starves the surface task and doesn't actually help volume. Surface and volume losses need to stay in approximate balance.

PR #142 closed as "vol_w=2.0 hurt surface badly; ratio imbalance". The natural follow-up — gentler weight that biases toward volume without crushing surface — has not been tested. That's this PR.

### CODE REVIEW

**Comment by morganmcg1** [comment]
ADVISOR: Closing PR #187 because All 8 training runs failed with zero metrics — no validation data to evaluate. This is part of a broader pattern of infrastructure failures in round 11 (see also fern PR #189 all 24 failed, frieren PR round10-compound-wd2e3 all 16 failed, alphonse PR round11-vol-pts-96k-clean crashed). The volume_loss_weight=1.5 hypothesis itself is scientifically sound — a gentler approach between baseline (1.0) and PR #142 (2.0) — but cannot be evaluated without working runs.

Closing to free nezuko's GPU for a fresh, higher-priority experiment. The hypothesis can be revisited in a future round with a confirmed working reproduce command verified against `python train.py --help`..

**Comment by morganmcg1** [comment]
STUDENT nezuko:

## Results

**Verdict: not a winner. Volume pressure improved marginally (−1.58%) but every other metric regressed ~+2–3%, with aggregate abupt_mean regressing +1.88%.**

### Test metrics (best-val checkpoint reload)

| Metric | This run (vol_w=1.5) | PR #115 SOTA (vol_w=1.0) | Δ | % change | AB-UPT |
|---|---:|---:|---:|---:|---:|
| **abupt_mean** | **10.779** | **10.580** | +0.199 | **+1.88%** | — |
| surface_pressure | 5.896 | 5.690 | +0.206 | +3.62% | 3.82 |
| wall_shear | 10.703 | 10.419 | +0.284 | +2.73% | 7.29 |
| **volume_pressure** | **12.539** | **12.740** | −0.201 | **−1.58%** ← target | 6.08 |
| tau_x | 9.112 | 8.908 | +0.204 | +2.29% | 5.35 |
| tau_y | 12.897 | 12.491 | +0.406 | +3.25% | 3.65 |
| tau_z | 13.450 | 13.071 | +0.379 | +2.90% | 3.63 |

Run completed cleanly: 9 epochs in 270.7 min, val→test ratio 1.117 (matches SOTA 1.115 — no instability artifacts).

### Per-epoch validation trajectory (`val_primary/*_rel_l2_pct`)

| ep | abupt | surf_p | vol_p | wall_shear | SOTA abupt ref |
|----|-------|--------|-------|------------|----------------|
| 0 | 59.765 | 47.846 | 36.325 | 65.306 | 53.75 |
| 1 | 26.089 | 18.532 | 14.279 | 29.298 | 24.15 |
| 2 | 17.437 | 12.014 | 9.620  | 19.609 | 16.51 |
| 3 | 13.815 | 9.335  | 7.766  | 15.525 | 13.47 |
| 4 | 12.099 | 8.038  | 7.027  | 13.591 | 11.83 |
| 5 | 11.044 | 7.231  | 6.503  | 12.420 | 10.88 |
| 6 | 10.368 | 6.704  | 6.237  | 11.631 | 10.16 |
| 7 | 9.918  | 6.359  | 6.012  | 11.119 | 9.73  |
| 8 | **9.653**  | 6.183  | 5.770  | 10.839 | 9.484 |

- ep0=59.77 exceeds the ≤55 watch threshold; ep1=26.09 exceeds the ≤25 threshold. Modest early destabilization from the rebalanced loss, consistent through training.
- Best-val (ep8) abupt: 9.653 vs SOTA 9.484 (+1.78% — tracks the test regression of +1.88%).
- Surface val at ep6 (6.704) and ep8 (6.183) stayed well below the 7.0 risk threshold — the *failure mode* is mild surface degradation, not the catastrophic ratio imbalance #142 hit at vol_w=2.0.

### Command

```bash
cd target/
torchrun --standalone --nproc_per_node=8 train.py \
  --agent nezuko \
  --optimizer lion --lr 1e-4 --weight-decay 5e-4 \
  --lion-beta1 0.9 --lion-beta2 0.99 \
  --no-compile-model --batch-size 4 --validation-every 1 \
  --epochs 9 \
  --train-surface-points 65536 --eval-surface-points 65536 \
  --train-volume-points 65536 --eval-volume-points 65536 \
  --model-layers 4 --model-hidden-dim 512 --model-heads 8 --model-slices 128 \
  --use-ema --ema-decay 0.999 \
  --lr-cosine-t-max 50 \
  --volume-loss-weight 1.5 \
  --wandb-group tay-round11-vol-loss-weight-1p5 \
  --wandb-name nezuko/round11-vol-loss-weight-1p5
```

### Run metadata

- W&B run (rank 0): `osbmpkmq` — group `tay-round11-vol-loss-weight-1p5`
- Total runtime: 270.7 min (matches SOTA's 287 min — single-flag loss-weight change, no compute change)
- Best epoch: 9 (last epoch)
- Peak GPU memory allocated: ~94 GiB / GPU (≈98% of H100 — same as SOTA, no compute change)

### What happened

**The hypothesis was directionally correct but the magnitudes don't add up to a win.**

Confirmed from this run:
1. **Volume pressure does respond to upweighting** — vol_p improved ~1.58% (12.74 → 12.54). The volume head gets more gradient signal under Lion's per-block sign update with vol_w=1.5, just as predicted.
2. **The reweighting tax on surface is real and proportional** — surface_pressure +3.62%, wall_shear +2.73%, all three tau axes +2.3–3.3%. The volume head doesn't free-ride; it steals from the surface head's effective gradient budget.
3. **The volume:surface trade-off is unfavorable at the aggregate** — gaining 0.2 absolute on volume costs ~0.2–0.4 absolute on each of six other axes, and abupt_mean (which averages the three primary l2_pcts) regresses +1.88%.

Compared to thorfinn's PR #142 (vol_w=2.0): #142 gave +10.78% abupt regression with +18% surface degradation and *flat* volume. This run gave +1.88% abupt regression with +2.7–3.6% surface degradation and a *real* 1.58% volume gain. So 1.5 sits in the regime where the volume signal is alive but isn't large enough to compensate for the surface tax — a smoother but still net-negative point on the same trade-off curve, not a different curve.

The post-mortem from #142 ("ratio imbalance with surface loss") generalizes: any `volume_loss_weight > 1.0` reweights gradient priority away from surface, and the marginal volume gain has diminishing returns relative to the surface cost. Closing the volume_pressure gap to AB-UPT (×2.06) likely requires an architectural mechanism (more volume capacity, better volume tokenization, longer training on volume-heavy batches), not a simple loss-scalar shift.

### Suggested follow-ups

1. **Architectural volume capacity** — give the volume head more parameters or its own deeper trunk, rather than fighting the shared backbone with a loss multiplier. This could let volume improve without surface degradation. Small targeted change: bump volume head's MLP from the shared `mlp_ratio` to e.g. 6 or 8 specifically on volume.
2. **Sub-1.0 surface loss weight, not super-1.0 volume weight** — `--surface-loss-weight 0.85 --volume-loss-weight 1.0` instead of `--surface-loss-weight 1.0 --volume-loss-weight 1.5`. Same ratio (1.5×), but this scaling shrinks the *combined* loss magnitude rather than enlarging it, which interacts differently with EMA and Lion's sign updates (smaller absolute gradient norms → less aggressive sign updates per step).
3. **Volume-only fine-tune phase at end of training** — last 1–2 epochs train only on volume with surface frozen. Captures any remaining volume signal without sacrificing the surface metrics already achieved.
4. **Per-batch volume sampling weight** — keep loss weights at 1.0 but oversample volume points (e.g. `--train-volume-points 98304 --train-surface-points 49152`). Increases gradient SNR on volume without changing the gradient ratio per parameter.
5. **Try `volume_loss_weight=1.25`** to fully sweep the regime between 1.0 and 1.5. If the test_primary degradation is roughly linear in (vol_w − 1.0), 1.25 should give ~half the surface tax (~+1%) for ~half the volume gain (~−0.8%) — still likely net negative, but it would confirm the linear trade-off and close out this lever.

**Comment by morganmcg1** [comment]
STUDENT nezuko:

Noting for the advisor: PR was closed due to reported infrastructure failures, but a successful run does appear to have completed. The results comment above (posted at 14:45 UTC) contains actual test metrics from W&B run `osbmpkmq` with full 9-epoch training (270.7 min). 

The run at logs/nezuko_pr187_volw_1p5_20260501_095141.log (started 09:51 UTC) ran to completion — best val 9.6528 at ep8, with test_primary metrics logged (abupt_mean: 10.779, volume_pressure: 12.539).

If those metrics are valid, the results are informative: vol_w=1.5 is net-negative on aggregate (−1.88% abupt regression) with a modest volume_pressure improvement (−1.58%). I defer to the advisor on whether to re-open or leave closed and use the data as a data point.
